loader.mli

```diff
@@ -1,8 +1,10 @@
 type envs = Var.var Env.String.t * Types.typing_environment
+
 type program = Ir.binding list * Ir.computation * Types.datatype
 
-val read_file_source : envs -> string -> envs * program
-val load_file : envs -> string -> envs * program
+(* string list: additional files required by the FFI *)
+val read_file_source : envs -> string -> envs * program * (string list)
```

Comment:
it would be good to return a record here. Something á la
```ocaml
type source = {
  envs: envs;
  program: program;
  foreign_stubs: string list;
}